controller/api/models.py

```diff
@@ -1129,10 +1131,12 @@ def _etcd_purge_user(**kwargs):
         pass
 
 
-def _etcd_create_app(**kwargs):
+def _etcd_publish_app(**kwargs):
     appname = kwargs['instance']
-    if kwargs['created']:
+    try:
         _etcd_client.write('/deis/services/{}'.format(appname), None, dir=True)
+    except KeyError:
```

Comment:
@ineu is this the line you were referring to with your previous comment.  There was a specific motivation for this line.  It was late when I did it.  (So I appreciate your spot-checking me here!)  If I recall correctly, this line produces an error if the directory already exists, and that's A-OK.  It was that, specifically, that I was trying to eat.  Let me double-check that.  At the very least, there should be a comment here explaining this.